### ISSUE TITLE
Generated instructions not discoverable by agent tools — no reference appended to instruction files

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#125]
## Summary

Running `npx llm-core-instructions` generates `.agents/linting-rules.md` but does not append a reference to it in any instruction file that agent tools actually read. The generated file is a dead letter — no Claude Code, Copilot, or OpenCode session will discover it without manual intervention.

## Problem

The generator writes a single file and exits:

```
.agents/linting-rules.md   ← written, never referenced
```

Meanwhile, every major agent tool discovers instructions from specific entry points:

| Agent tool | Reads | Generator touches? |
|---|---|---|
| Claude Code | `CLAUDE.md` | ❌ |
| GitHub Copilot | `.github/copilot-instructions.md` | ❌ |
| OpenCode / generic | `AGENTS.md` | ❌ |

None of these files get a reference to `.agents/linting-rules.md` after generation. The user must manually add one — but the README doesn't mention this step, so most users won't.

## Proposed Enhancement

After writing `.agents/linting-rules.md`, the CLI should detect which Markdown instruction files exist in the project and append a reference line to each one.

### Scope: repo-root Markdown instruction files only

The generator targets only the well-known instruction files at the repository root:

- `<repo-root>/AGENTS.md`
- `<repo-root>/CLAUDE.md`
- `<repo-root>/.github/copilot-instructions.md`

This is deliberately limited to repo-root locations. Agent tools may support additional locations or nearest-file lookup behavior (e.g. nested `CLAUDE.md` files), but those are not in scope.

**Out of scope:** `.cursor/rules/` is a directory-based format, not a single Markdown file. Cursor injection requires a separate contract and is not part of this issue.

### CLI vs. library boundary

Injection belongs to the **CLI path only** (`npx llm-core-instructions`). The programmatic API `generateInstructions()` must remain pure — it returns `content` and rule metadata and does not mutate repo files. No new file-writing responsibilities are added to the library export.

### Detection & injection logic

1. After writing `.agents/linting-rules.md`, check for the presence of known Markdown instruction files at the repo root
2. For each file found, append a delimited reference block with the correct relative path
3. Skip files that don't exist (don't create them)
4. Use delimited comment markers so the block is idempotent — re-running replaces in place rather than duplicating

### Per-target relative path handling

The injected block must use the correct relative path from the target file to `.agents/linting-rules.md`:

| Target file | Relative path |
|---|---|
| `AGENTS.md` | `.agents/linting-rules.md` |
| `CLAUDE.md` | `.agents/linting-rules.md` |
| `.github/copilot-instructions.md` | `../.agents/linting-rules.md` |

The implementation must compute the relative path per target file rather than hardcoding a single path.

### Appended block examples

For root-level files (`AGENTS.md`, `CLAUDE.md`):

```markdown
<!-- llm-core-instructions:start -->
See [`.agents/linting-rules.md`](.agents/linting-rules.md) for coding guidelines derived from your ESLint config.
Regenerate with: `npx llm-core-instructions`
<!-- llm-core-instructions:end -->
```

For `.github/copilot-instructions.md`:

```markdown
<!-- llm-core-instructions:start -->
See [`.agents/linting-rules.md`](../.agents/linting-rules.md) for coding guidelines derived from your ESLint config.
Regenerate with: `npx llm-core-instructions`
<!-- llm-core-instructions:end -->
```

The `<!-- llm-core-instructions:start/end -->` markers make it:
- **Idempotent** — re-running replaces the block instead of appending duplicates
- **Non-destructive** — doesn't touch any other content in the file

### Flags

| Flag | Description |
|---|---|
| `--no-inject` | Generate `.agents/linting-rules.md` only, skip appending references (previous default behavior) |

Default behavior changes to inject references. This is a **product decision** for out-of-the-box discoverability, not just an implementation detail. `--no-inject` restores the old behavior for users who prefer manual wiring.

**Deferred:** `--clean` (strip injected blocks from instruction files) is not in scope for this issue. It can be addressed separately if needed.

### `--dry-run` semantics

`--dry-run` writes **nothing** — no `.agents/linting-rules.md` and no injected blocks. It prints the generated markdown to stdout only. This is consistent with the existing contract and must remain so.

### Console output

On successful injection, the CLI prints one line per modified file, matching the existing output style (`Generated <path>`):

```
Generated .agents/linting-rules.md
Injected reference into AGENTS.md
Injected reference into .github/copilot-instructions.md
```

If no instruction files are found, print the generated file line only (no additional message needed).

### Injection behavior
- [ ] CLI detects existing repo-root Markdown instruction files (`AGENTS.md`, `CLAUDE.md`, `.github/copilot-instructions.md`)
- [ ] Computes correct relative path per target file (`.agents/linting-rules.md` for root files, `../.agents/linting-rules.md` for `.github/`)
- [ ] Appends delimited reference block to each detected file
- [ ] Preserves all surrounding file content unchanged
- [ ] Re-running replaces the delimited block in place (no duplication)
- [ ] Skips missing files without creating them
- [ ] If a target file exists but is not writable, log a warning to stderr and continue (do not crash)
- [ ] `generateInstructions()` library API is unchanged — no file mutation

### Flag behavior
- [ ] Default (no flags): generates file + injects references
- [ ] `--no-inject`: generates file only, no injection (previous default)
- [ ] `--dry-run`: writes nothing to disk (no generated file, no injected blocks)
- [ ] `--dry-run` + `--no-inject`: equivalent to `--dry-run` alone

### Tests
- [ ] Surrounding file content preserved after injection
- [ ] Delimited block replaced in place without duplication
- [ ] Missing instruction files skipped without error
- [ ] Correct relative path used for each target file (root vs. `.github/`)
- [ ] Writable-file error handled gracefully (warning to stderr, no crash)
- [ ] `--dry-run` produces stdout output but writes nothing
- [ ] `--no-inject` skips injection entirely
- [ ] `--dry-run` + `--no-inject` behaves correctly
- [ ] Library API (`generateInstructions()`) unchanged — existing tests pass

### Documentation
- [ ] README updated to document the new default behavior and flags
- [ ] README updated to reflect that injection is automatic (not manual)

## Context

- Related: #122 (edge cases in the same feature)
- Source: manual investigation of `src/cli/generate-instructions.ts` and `src/instructions/generator.ts`
